apps/vast/pyvast_threatbus/message_mapping.py

```diff
@@ -1,107 +1,100 @@
 from dateutil import parser as dateutil_parser
 from ipaddress import ip_address
 import json
-from threatbus.data import Intel, IntelType, Sighting
-
-to_vast_intel = {
-    IntelType.IPSRC: "ip",
-    IntelType.IPDST: "ip",
-    IntelType.HOSTNAME: "domain",
-    IntelType.DOMAIN: "domain",
-    IntelType.DOMAIN_IP: "domain",
-    IntelType.URL: "url",
-    IntelType.URI: "url",
+from stix2 import Indicator, Sighting
+from threatbus.data import ThreatBusSTIX2Constants
+from threatbus.stix2_helpers import is_point_equality_ioc, split_object_path_and_value
+from typing import Tuple, Union
+
+vast_ioc_type_map = {
+    "ipv4-addr:value": "ip",
+    "ipv6-addr:value": "ipv6",
```

Comment:
I think I know why this works now despite the mismatch between vasts builtin `addr` type and these: The function `indicator_to_vast_query` maps these to value predicates and the type name is not used.

In case you plan to make use of this `ip` vs `ipv6` separation it should be documented why it exists, otherwise you could just use `addr` for both to align with vasts builtin type.

Reply:
This naming refers to VAST matcher IoC types, see here: https://docs.tenzir.com/vast/features/threat-intel-matching#adding-iocs

When building VAST *retro-queries* we don't use the ioc_type, only the values.

Reply:
Ah yes, the matcher treats these differently, sorry for the confusion.